src/AppBundle/Controller/RestaurantController.php

```diff
@@ -261,9 +261,29 @@ public function indexAction($id, $slug, Request $request, SlugifyInterface $slug
 
             $numberOfVotes = count($restaurant->getPledge()->getVotes());
 
+            $userId = $this->getUser()->getId();
```

Comment:
This whole piece of code checking if a user has already voted could be refactored by adding a method to the `Pledge` class. 

```php
public function hasVoted(UserInterface $user): bool
{
  // Loop on votes
  // If a vote has the same $user, return true
  // Else return false
}
```